### PR TITLE
chore(datepicker): QoL improvements 

### PR DESCRIPTION
**Pull request by joseazevedo88**
Add prevent open on focus optional prop and onKeyDown callback to control open/close with ref

### CODE REVIEW

File: src/components/DatePicker/DatePicker.stories.tsx [hunk 1]
```diff
@@ -115,6 +116,13 @@ const TemplateDateRange: StoryFn<DatePickerProps> = () => {
             minDate={subDays(new Date(), 40)}
             maxDate={new Date()}
             variant="range"
+            ref={testRef}
```
Comment (imoutaharik) [inline src/components/DatePicker/DatePicker.stories.tsx:119]: Can we change the name testRef?

**Review by imoutaharik — COMMENTED** [review]
Looks good to me.

I just noticed something about the keyboard navigation, when the range is big (2 months for example) and you tab on the Datepicker, it's kind of hard to interact with it.
You need to use the << <  > >> buttons to find the initial date so you can finally interact with the dates.

File: src/components/DatePicker/DatePicker.stories.tsx (hunk 1 above)
Comment (joseazevedo88) [reply src/components/DatePicker/DatePicker.stories.tsx]: Sure, leftover

**Comment by joseazevedo88** [comment]
> Looks good to me.
> 
> I just noticed something about the keyboard navigation, when the range is big (2 months for example) and you tab on the Datepicker, it's kind of hard to interact with it. You need to use the << < > >> buttons to find the initial date so you can finally interact with the dates.

Yup, added a "hacky" way of doing it in the stories, manipulating the ref, as I'll need it also for clarify. The lib is somewhat outdated and doesn't seem to be maintained anymore, some things are not up to date in terms of react standards.

**Review by imoutaharik — APPROVED** [review]

**Pull request merged**